coderd/userauth.go

```diff
@@ -95,7 +98,7 @@ func (api *API) userOAuth2Github(rw http.ResponseWriter, r *http.Request) {
 	}
 
 	// The default if no teams are specified is to allow all.
-	if len(api.GithubOAuth2Config.AllowTeams) > 0 {
+	if len(api.GithubOAuth2Config.AllowOrganizations) > 0 && len(api.GithubOAuth2Config.AllowTeams) > 0 {
```

Comment:
This would silently allow all logins even if teams are set but organizations are not. I'd say that's a configuration error, but we shouldn't silently allow it.

We should perhaps keep this `if` as it was and recheck that `selectedMembership` is non-nil.